addons/vehicles/XEH_postInit.sqf

```diff
@@ -1,7 +1,42 @@
 // by esteldunedain
 #include "script_component.hpp"
 
+// eject destroyed vehicle init
+[QGVAR(ejectDestroyed),{
+    TRACE_2("ejectDestroyed event",_this,typeOf _this);
+    _this removeEventHandler ["HandleDamage", _this getVariable QGVAR(ejectDestroyedHDEH)];
+    _this setVariable [QGVAR(ejectDestroyedHDEH), nil];
+}] call CBA_fnc_addEventHandler;
+private _ejectDestroyedClasses = [];
+{
+    private _ejectDestroyedClass = configName _x;
+    if (-1 == _ejectDestroyedClasses findIf {_ejectDestroyedClass isKindOf _x}) then {
+        TRACE_1("init ejectDestroyed class",_ejectDestroyedClass);
+        _ejectDestroyedClasses pushBack _ejectDestroyedClass;
+        [_ejectDestroyedClass, "init", {
+            params ["_vehicle"];
+            if (!alive _vehicle) exitWith {};
+            TRACE_2("init ejectDestroyed vehicle",_vehicle,typeOf _vehicle);
+            private _eh = _vehicle addEventHandler ["HandleDamage", {
+                params ["_vehicle"];
+                if (!alive _vehicle) then {
+                    TRACE_1("ejectDestroyed ejectDestroyedHDEH",_this);
+                    {
+                        if (alive _x) then {
+                            moveOut _x;
+                        };
+                    } forEach crew _vehicle;
+                    [QGVAR(ejectDestroyed), _vehicle] call CBA_fnc_globalEvent;
+                };
+            }];
+            _vehicle setVariable [QGVAR(ejectDestroyedHDEH), _eh];
```

Comment:
You don't need to store this variable, just use `_thisEventHandler` in the event to remove itself

Reply:
I need it because `HandleDamage` is local and doesn't hit on all machines, so I don't have `_thisEventHandler` value everywhere

Reply:
🤔 Huh? Of course you have `_thisEventHandler` everywhere, it's a block of code used in an event handler added with `addEventHandler`.

Personally I would much rather have some leftover event handlers on some machines which will remove themselves if a dead vehicle ever changes locality and takes a single hit of damage again (which, let's be honest, isn't likely). It will make the code more elegant and remove unnecessary network traffic.

Reply:
But why do you need it? https://github.com/acemod/ACE3/pull/6330/files#diff-71d8e1c52a77144d92094e89a77b7a55R6
That?
The eventhandler and variables are removed anyway when the vehicle is destroyed aren't they?

Reply:
@SilentSpike Ref: https://github.com/acemod/ACE3/pull/6330#discussion_r186632443

Reply:
I have `_thisEventHandler` only where vehicle is local. EH is fired only on one machine so I have to clean up EHs on all other machines. I always thought cleanup is necessary. If you insist I can get rid of it, no problem.

> The eventhandler and variables are removed anyway when the vehicle is destroyed aren't they?

No, they aren't removed.

Reply:
They will clean themselves up in the event that a dead vehicle switches locality and receives more damage (again, not likely). It's totally harmless to leave them sitting there doing nothing.